packages/astro/src/vite-plugin-config-alias/index.ts

```diff
@@ -33,10 +33,9 @@ const getConfigAlias = (settings: AstroSettings): Alias[] | null => {
 					.join('')}$`,
 			);
 
-			/** Internal index used to calculate the matching id in a replacement. */
-			let matchId = 0;
-
 			for (const value of values) {
+				/** Internal index used to calculate the matching id in a replacement. */
+				let matchId = 0;
```

Comment:
This is the fix. The index needs resetting between paths, because it's per-regex